linkml/generators/pydanticgen/template.py

```diff
@@ -454,6 +523,17 @@ def __add__(self, other: Import) -> "Imports":
                         merged = e.merge(other)
                         imports.extend(merged)
                         break
+
+        # SPECIAL CASE - __future__ annotations must happen at the top of a file
+        # TODO: Make a sort method
+        futures = []
+        for item in imports[-2:]:
+            if item.module == "__future__":
+                futures.append(item)
+        if len(futures) > 0:
+            imports.remove(futures[0])
+            imports.insert(0, futures[0])
+
```

Comment:
Could you describe what this is doing more in the comments? It looks like it looks searches the last two imports for imports from `__future__` and if any are found, it moves the first import from `__future__` to the beginning of the imports list. But why search only the last two imports and why move only the first of those?

Reply:
ah right. so there can only ever be 2 imports added at this stage in the method - usually there will only be one, but we can get two depending on if we are getting a conditional import and already have one.

`__future__` imports shouldn't be a conditional import, though, and should always merge down into a single import object, but it might be in the second to last or last position.

now that i'm looking at this though, i agree it's hideous, and i should just use `sorted` with a lambda like

`imports = sorted(imports, lambda i: i.module == '__future__', reverse=True)`

i'll do that after i merge your comments

Reply:
here ya go. "a single line takes 30 lines to test" should be some kind of folk idiom or something.

https://github.com/linkml/linkml/pull/1887/commits/df0c26714a540cfa4d83efd11b22feb21bf16755

Reply:
Thanks! Looks great and much more understandable

Reply:
I always think I am decreasing the proportion of lines I write that I later look back on and can say nothing but "what the fuck," but moments like this remind me its more of an asymptote to a long term average than trending to zero